Edge max charge 1

adamnovak opened the pull request:
Only charge 1 for passing up any number of alternative edges.

In order to make the new semantics work, I re-did some of the prev/next kpaths of node logic.

In order to make vg build, I have had to update it to the latest protobuf. This involved doing some casting, since protobuf changed their 64-bit integer type and it no longer is the same as int64_t, at least on my platform.

Depends on https://github.com/ekg/xg/pull/8

adamnovak commented:
Note that Travis won't be able to test this until the submodule commits I'm referencing get merged into the submodules.

ekg commented:
Re: casting. We might want to move from overuse of `size_t` and `int64_t` to locally-defined types for sizes and ids. `id_type` and `size_type`. I've been thinking about a find and replace to fix it up.

ekg commented:
Which compiler are you using? Have you been able to build on OSX? @edawson has been trying to build on mac and failing...

glennhickey commented:
I just successfully built this on my mac for what it's worth...

On Tue, Oct 27, 2015 at 8:39 AM, Erik Garrison <notifications@github.com>
wrote:

> Merged #131 <https://github.com/ekg/vg/pull/131>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ekg/vg/pull/131#event-446714580>.
>

ekg commented:
What version are you running?

glennhickey commented:
For vg, I just rebased on your master, updated submodules, then did make
clean ; make.

For my mac, I think it's running the latest everything:  OS X 10.11.1 with
whichever XCode came with it, as well as an up-to-date MacPorts..

On Tue, Oct 27, 2015 at 10:03 AM, Erik Garrison <notifications@github.com>
wrote:

> What version are you running?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ekg/vg/pull/131#issuecomment-151505504>.
>

adamnovak commented:
FWIW, my compiler is:

```
[anovak@kolossus ~]$ g++ --version
g++ (GCC) 4.9.0
Copyright (C) 2014 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

On Tue, Oct 27, 2015 at 7:13 AM, Glenn Hickey <notifications@github.com>
wrote:

> For vg, I just rebased on your master, updated submodules, then did make
> clean ; make.
>
> For my mac, I think it's running the latest everything: OS X 10.11.1 with
> whichever XCode came with it, as well as an up-to-date MacPorts..
>
> On Tue, Oct 27, 2015 at 10:03 AM, Erik Garrison <notifications@github.com>
> wrote:
>
> > What version are you running?
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <https://github.com/ekg/vg/pull/131#issuecomment-151505504>.
>
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ekg/vg/pull/131#issuecomment-151508807>.
>
